src/components/home/SearchModal.tsx

```diff
@@ -0,0 +1,115 @@
+import { ArrowLeftBlackIcon, FlightGrayIcon } from '@/assets/svg';
+import Button from '@/components/common/Button.tsx';
+import InputForm from '@/components/common/InputForm.tsx';
+import CityGroup from '@/components/home/CityGroup.tsx';
+import { cityGroups } from '@/pages/home/cityGroups.ts';
+import { FogEffect, flexCssGenerator } from '@/styles/customStyle.ts';
+import React, { useEffect, useRef, useState } from 'react';
+import styled from 'styled-components';
+
+interface props {
+	isOpen: boolean;
+	onModalToggle: () => void;
+}
+
+const SearchModal = ({ isOpen = false, onModalToggle }: props) => {
+	const [isButtonClicked, setIsButtonClicked] = useState(false);
+	const [showFogEffect, setShowFogEffect] = useState(true);
+	const scrollRef = useRef<HTMLDivElement>(null);
+
+	const onSearchClick = () => {
+		onModalToggle();
+	};
+
+	const onOkinawaClicked = (state: boolean) => {
+		setIsButtonClicked(state);
+	};
```

Comment:
p2: 해당 이벤트 핸들러 네이밍을 `onOkinawaClicked`보다는 네이밍 컨벤션에 맞추어 `handleOkinawaClick`을 사용하는 것이 어떨까요? 또한, 클릭 시 상태가 직곽적이지 못한 것 같아. 다음과 같이 수정하는 것은 어떨까요?
```suggestion
	const handleOkinawaClick = () => {
                setIsButtonClicked((prev) => !prev);
};
```

Reply:
오 좋습니다 !!
요거 함수 onClick 으로 아예 변경 될 예정으로 로직이 아예 변경되었어유